### ISSUE TITLE
Using `layout:` in `*.mdx` causes build to crash

### ISSUE THREAD

**Issue by ota-meshi** [withastro/astro#10502]
### Astro Info

```block
❯ astro info
Astro                    v4.5.6
Node                     v18.18.0
System                   Linux (x64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             @astrojs/mdx
```

### Describe the Bug

If you use `layout:` in a `*.mdx` file like the following, the `astro build` command will generate an error and exit.

```mdx
---
layout: ../Layout.astro
---

# Hello world!
```

```
> @example/with-mdx@0.0.1 build
> astro build

12:38:39 [build] output: "static"
12:38:39 [build] directory: /home/projects/github-razdlq/dist/
12:38:39 [build] Collecting build info...
12:38:39 [build] ✓ Completed in 259ms.
12:38:39 [build] Building static entrypoints...
12:38:40 [ERROR] [vite] x Build failed in 1.10s
src/pages/index.mdx (3:15): Identifier "_jsx" has already been declared
file: /home/projects/github-razdlq/src/pages/index.mdx:3:15
1: ---
2: layout: ../Layout.astro
3: ---
      ^
4: 
5: # Hello world!
  Stack trace:
    at getRollupError (/home/projects/github-razdlq/node_modules/rollup/dist/es/shared/parseAst.js:527:41)
    at Module.error (/home/projects/github-razdlq/node_modules/rollup/dist/es/shared/node-entry.js:13679:29)
    at ImportDeclaration.initialise (/home/projects/github-razdlq/node_modules/rollup/dist/es/shared/node-entry.js:10143:28)
    at convertNodeList (/home/projects/github-razdlq/node_modules/rollup/dist/es/shared/node-entry.js:12931:34)
    at convertNode (/home/projects/github-razdlq/node_modules/rollup/dist/es/shared/node-entry.js:12922:28)
```

If you put a breakpoint here and look at the `result.code` after `babel.transformAsync`,

https://github.com/withastro/astro/blob/2fc7231df28e5a3425ee47b871ba3766e0856bd8/packages/astro/src/vite-plugin-mdx/index.ts#L36

you will see that `import` is duplicated.

```js
import { __astro_tag_component__ } from "astro/runtime/server/index.js";
import { Fragment, jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "astro/jsx-runtime";
import { jsx as layoutJsx, jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from "astro/jsx-runtime";
const MDXLayout = async function ({
  children
}) {
//...
```

I'm not sure which babel plugin it's a problem with.

### What's the expected result?

The build is successful.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-razdlq?file=src%2Fpages%2Findex.mdx

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by wonrax** [comment]
I'm having the same error after performing `npm update` today
```json
"@astrojs/mdx": "^2.2.0",
"@astrojs/react": "^3.1.0",
"astro": "^4.5.5",
```

**Comment by ota-meshi** [comment]
It seems like it's a babel issue. The results are different between v7.24.0 and v7.24.2. I still don't know which package of babel is the problem.

[babel repl v7.24.0](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAbzgMSgQwOYgKYDt5oDOcAZhBHAL6lQQhwBEU2aAxjAPQBWhAHgLRQArvlDYGAbgBQoSLEQp0WPAWIAjNFCo06jZm048Bw0TklTWEXIQJwAvHAA8APkcdnUuHCA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.24.0&externalPlugins=&assumptions=%7B%7D)
[babel repl v7.24.2](https://babeljs.io/repl/#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=JYWwDg9gTgLgBAbzgMSgQwOYgKYDt5oDOcAZhBHAL6lQQhwBEU2aAxjAPQBWhAHgLRQArvlDYGAbgBQoSLEQp0WPAWIAjNFCo06jZm048Bw0TklTWEXIQJwAvHAA8APkcdnUuHCA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2&prettier=false&targets=&version=7.24.2&externalPlugins=&assumptions=%7B%7D)

**Comment by ota-meshi** [comment]
I submitted a PR to babel.

https://github.com/babel/babel/pull/16370

**Comment by nicolo-ribaudo** [comment]
I'm releasing a new Babel version, it should be out in ~15 mins

**Comment by nicolo-ribaudo** [comment]
Released as 7.24.3

**Comment by Princesseuh** [comment]
Made a PR to ensure we're using the latest version! Thank you all for working on this so quickly, awesome work.
